plugins/webapi/tools/message/send_message.go

```diff
@@ -0,0 +1,96 @@
+package message
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/iotaledger/hive.go/events"
+	"github.com/labstack/echo"
+
+	"github.com/iotaledger/goshimmer/packages/jsonmodels"
+	"github.com/iotaledger/goshimmer/packages/tangle"
+)
+
+const maxIssuedAwaitTime = 5 * time.Second
+
+func SendMessage(c echo.Context) error {
+	if !deps.Tangle.Synced() {
+		return c.JSON(http.StatusBadRequest, jsonmodels.DataResponse{Error: tangle.ErrNotSynced.Error()})
+	}
+	deps.Tangle.Booker.Lock()
+	defer deps.Tangle.Booker.Unlock()
+
+	if c.Request().Body == nil {
+		return c.JSON(http.StatusBadRequest, jsonmodels.DataResponse{Error: "invalid message, error: request body is missing"})
+	}
+
+	var request jsonmodels.DataRequest
+	if err := c.Bind(&request); err != nil {
+		return c.JSON(http.StatusBadRequest, jsonmodels.DataResponse{Error: err.Error()})
+	}
+	if len(request.Data) == 0 {
+		return c.JSON(http.StatusBadRequest, jsonmodels.DataResponse{Error: "no data provided"})
+	}
+
+	msg, _, err := tangle.MessageFromBytes(request.Data)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, jsonmodels.DataResponse{Error: fmt.Sprintf("error: %s", err.Error())})
+	}
+
+	if msg.Version() != tangle.MessageVersion {
```

Comment:
Why should user even care if he puts a correct version in the API request if the message is prepared by node anyway?

Reply:
yeah. We shouldn't care about this.